### ISSUE TITLE
Persist the state of kube-spawn

### ISSUE THREAD

**Issue by robertgzr** [kinvolk/kube-spawn#137]
kube-spawn has to do the following steps on every execution of the `setup` command:
- write setup scripts and configuration files
- compose mount arguments

The generated scripts should be written to `<kube-spawn-dir>/default/scripts/` 
and information needed to bring up the cluster should go into a file e.g. `<kube-spawn-dir>/default/spawnfile`, including:
- mount options 
- k8s version and runtime
- number of nodes

This way they can be reused on subsequent restarts of the cluster.

A `destroy` command would clean up and make `kube-spawn` regenerate ^

Related: #37 #38

**Comment by robertgzr** [comment]
Can https://godoc.org/github.com/coreos/go-systemd/machine1 be used to gather node details from just the machine name that we get from the node count?

**Comment by dongsupark** [comment]
Yes, I think in general that is the way to go in the long term. I haven't thought about using go-systemd though.

**Comment by robertgzr** [comment]
I am putting together a central config object that holds all the variables needed to start/init/stop/destroy the cluster:

```
type ClusterConfiguration struct {
	KubeSpawnDir string
	Name string
	ContainerRuntime string
	KubernetesVersion string
	DevCluster bool
	Nodes int
	Image string
	RuntimeConfiguration        // holds runtime binary paths
	Bindmounts                  // holds additional bind mounts
	Machines map[string]string  // map machine name to ip address
	Token string                // kubeadm token for join
}
```

**Comment by robertgzr** [comment]
Resolved with #161
